pkg/bootstrap/platform/gcp.go

```diff
@@ -173,11 +203,16 @@ func IsGCP() bool {
 // Metadata returned by the GCP Environment is taken from the GCE metadata
 // service.
 func NewGCP() Environment {
+	zone := GCPStaticMetadata[GCPZone]
+	if zone == "" {
+		zone = zoneFromResolvConf()
+	}
+
 	return &gcpEnv{
 		fillMetadata: lazy.New(func() (bool, error) {
 			return shouldFillMetadata(), nil
 		}),
-		cachedZone: atomic.NewString(GCPStaticMetadata[GCPZone]),
+		cachedZone: atomic.NewString(zone),
```

Comment:
nit: we could probably make it lazier by moving it into `getPodZone`.

Reply:
done!